src/commands/cmd_delete.c

```diff
@@ -25,10 +25,8 @@ int MGraph_Delete(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
 
 	RedisModuleString *graph_name = argv[1];
 	GraphContext *gc = GraphContext_Retrieve(ctx, graph_name, false, false);    // Increase ref count.
-	if(!gc) {
-		RedisModule_ReplyWithError(ctx, "Graph is either missing or referred key is of a different type.");
-		goto cleanup;
-	}
+	// If the GraphContext is null, key access failed and an error has been emitted.
+	if(!gc) return REDISMODULE_ERR;
```

Comment:
Don't we need to free the QueryContext? `QueryCtx_Free` ?

Reply:
In this case we don't, because the only thing we've written to the QueryCtx is a query start time, and the container itself is thread-specific and reused.

We could add such a call for future-proofing, but since automation would catch any errors, I'm not overly concerned!